core/app/models/spree/promotion_action.rb

```diff
@@ -19,5 +19,15 @@ class PromotionAction < Spree::Base
     def perform(options = {})
       raise 'perform should be implemented in a sub-class of PromotionAction'
     end
+
+    def duplicate
+      duplicated_action = self.dup
+      duplicate_calculator(duplicated_action) if try(:calculator)
```

Comment:
Actions only have calculators if the subclass includes a module (`Spree::CalculatedAdjustments`), so I figured it would be safer to not assume the subclass includes that.